colossalai/inference/tensor_parallel/engine.py

```diff
@@ -111,37 +196,62 @@ def _prepare_with_shard_config(self, shard_config: Optional[ShardConfig] = None)
-        model_name = model.__class__.__name__
+        model_name = self.model.__class__.__name__
         assert model_name in self.supported_models, f"Unsupported model cls {model_name} for TP inference."
-        policy = get_autopolicy(model, inference_only=True)
-        self.model, _ = shardformer.optimize(model, policy)
-        self.model = self.model.cuda()
+        if model_name == "LlamaForCausalLM":
+            init_to_get_rotary(self.model.model, base=10000)
+        policy = get_autopolicy(self.model, inference_only=True)
+        self.sharded_model, _ = shardformer.optimize(self.model, policy)
+        self.sharded_model = self.sharded_model.cuda()
 
     @property
     def supported_models(self) -> List[str]:
-        return _supported_models
+        return list(_supported_models.keys())
 
-    def generate(self, input_tokens: Union[BatchEncoding, dict, list, torch.Tensor], **generate_kwargs) -> torch.Tensor:
+    def generate(self,
+                 prompts: Optional[Union[str, List[str]]] = None,
+                 prompt_token_ids: Optional[List[List[int]]] = None,
+                 **generate_kwargs) -> Union[List[RequestOutput], torch.Tensor]:
         """Generate token sequence.
 
         Args:
-            input_tokens: could be one of the following types
-                1. BatchEncoding or dict (e.g. tokenizer batch_encode)
-                2. list of input token ids (e.g. appended result of tokenizer encode)
-                3. torch.Tensor (e.g. tokenizer encode with return_tensors='pt')
+            prompts: A list of prompts to generate completions for.
+            prompt_token_ids: A list of token IDs for the prompts. If None, we
+                use the tokenizer to convert the prompts to token IDs.
         Returns:
             torch.Tensor: The returned sequence is given inputs + generated_tokens.
         """
+
+        if prompts is None and prompt_token_ids is None:
+            raise ValueError("Either prompts or prompt_token_ids must be "
+                             "provided.")
+        if isinstance(prompts, str):
+            # Convert a single prompt to a list.
+            prompts = [prompts]
+        if prompts is not None and prompt_token_ids is not None:
+            if len(prompts) != len(prompt_token_ids):
+                raise ValueError("The lengths of prompts and prompt_token_ids "
+                                 "must be the same.")
+
+        if self.use_continous_batching:
+            sampling_params = SamplingParams(temperature=0.0, max_tokens=self.max_output_len)
+            return self.llm_engine.generate(prompts=prompts,
+                                            prompt_token_ids=prompt_token_ids,
+                                            sampling_params=sampling_params)
+
+        if prompt_token_ids is None:
+            input_tokens = self.tokenizer.batch_encode_plus(prompts, return_tensors="pt", padding=True)
+        else:
+            input_tokens = prompt_token_ids
```

Comment:
Where to place the attention mask for the condition that prompt_token_ids is not None?

Reply:
The following is the relevant logic for processing attention mask. I reused the relevant processing logic from the previous engine.